src/virtual-source-buffer.js

```diff
@@ -410,6 +396,19 @@ export default class VirtualSourceBuffer extends videojs.EventTarget {
       return segmentObj;
     }, sortedSegments);
 
+    // Create the real source buffers if they don't exist by now
+    if (!this.videoBuffer_ && !this.audioBuffer_) {
+      if (sortedSegments.video.bytes === 0) {
+        this.videoCodec_ = null;
```

Comment:
maybe add a comment stating something like "If we don't have any video data here then we want to ignore the codec, if we have one, as its wrong"